doc/dev/developer_guide/essentials.rst

```diff
@@ -165,41 +163,54 @@ See instructions at :doc:`/install/build-ceph`.
 
 Using ccache to speed up local builds
 -------------------------------------
+`ccache`_ can make the process of rebuilding the ceph source tree faster. 
+
+Old build artifacts can cause build failures. You might introduce these
+artifacts unknowingly when switching from one branch to another. If you have
+experienced a build failure, you might use `ccache`_ to build a full, clean
```

Comment:
this last sentence doesn't make sense - you'd clean the source tree as described below if you encounter build failures. Using ccache is irrelevant.